django/db/models/fields/related_lookups.py

```diff
@@ -98,49 +75,31 @@ def get_prep_lookup(self):
         return super().get_prep_lookup()
 
     def as_sql(self, compiler, connection):
-        if isinstance(self.lhs, MultiColSource):
+        if isinstance(self.lhs, Cols):
             # For multicolumn lookups we need to build a multicolumn where clause.
             # This clause is either a SubqueryConstraint (for values that need
             # to be compiled to SQL) or an OR-combined list of
             # (col1 = val1 AND col2 = val2 AND ...) clauses.
-            from django.db.models.sql.where import (
-                AND,
-                OR,
-                SubqueryConstraint,
-                WhereNode,
-            )
-
-            root_constraint = WhereNode(connector=OR)
+            from django.db.models.sql.where import SubqueryConstraint
+
             if self.rhs_is_direct_value():
                 values = [get_normalized_value(value, self.lhs) for value in self.rhs]
-                for value in values:
-                    value_constraint = WhereNode()
-                    for source, target, val in zip(
-                        self.lhs.sources, self.lhs.targets, value
-                    ):
-                        lookup_class = target.get_lookup("exact")
-                        lookup = lookup_class(
-                            target.get_col(self.lhs.alias, source), val
-                        )
-                        value_constraint.add(lookup, AND)
-                    root_constraint.add(value_constraint, OR)
+                lookup_cls = tuple_lookups["in"]
+                lookup = lookup_cls(self.lhs, values)
+                return lookup.as_sql(compiler, connection)
```

Comment:
I'd swap to `compiler.compile` instead to allow future `as_vendor` implementations

```suggestion
                return compiler.compile(lookup)
```